### DISCUSSION TITLE
TypeError: validation_step() missing 1 required positional argument: 'dataloader_nb'

### DISCUSSION THREAD

**Discussion by MooreManor** [Lightning-AI/pytorch-lightning#14238] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
```python
def validation_step(self, batch, batch_nb, dataloader_nb, vis=False, save=True, mesh_save_dir=None):
```

My pytorch-lightning version is 1.6.0, but the code is written with pytorch-lightning 1.1.8. And when I ran the code, I met the bug `validation_step() missing 1 required positional argument: 'dataloader_nb'`.

How can I  fix it and pass parameters to `validation_step() `?

**Comment by rohitgr7** [comment, marked as answer]
if you are using a single val dataloader:
```py
def validation_step(self, batch, batch_idx):
```

if you are using multiple val dataloader:
```py
def validation_step(self, batch, batch_idx, dataloader_idx):
```

for consistency, you can just do
```py
def validation_step(self, batch, batch_idx, dataloader_idx=0):
```
